vcr/issuer/networkPublisher.go

```diff
@@ -0,0 +1,132 @@
+package issuer
+
+import (
+	"encoding/json"
+	"fmt"
+	ssi "github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/go-did/vc"
+	"github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/network"
+	"github.com/nuts-foundation/nuts-node/network/transport"
+	"github.com/nuts-foundation/nuts-node/vcr/credential"
+	"github.com/nuts-foundation/nuts-node/vcr/log"
+	"github.com/nuts-foundation/nuts-node/vdr/doc"
+	vdr "github.com/nuts-foundation/nuts-node/vdr/types"
+)
+
+type networkPublisher struct {
+	networkTx       network.Transactions
+	didDocResolver  vdr.DocResolver
+	serviceResolver doc.ServiceResolver
+	keyResolver     keyResolver
+}
+
+// VcDocumentType holds the content type used in network documents which contain Verifiable Credentials
+const VcDocumentType = "application/vc+json"
+
+// NewNetworkPublisher creates a new networkPublisher which implements the Publisher interface.
+// It is the default implementation to use for issuers to publish credentials and revocations to the Nuts network.
+func NewNetworkPublisher(networkTx network.Transactions, docResolver vdr.DocResolver, keyResolver crypto.KeyResolver) Publisher {
+	return &networkPublisher{
+		networkTx:       networkTx,
+		didDocResolver:  docResolver,
+		serviceResolver: doc.NewServiceResolver(docResolver),
+		keyResolver: vdrKeyResolver{
+			docResolver: docResolver,
+			keyResolver: keyResolver,
+		},
+	}
+
+}
+
+func (p networkPublisher) PublishCredential(verifiableCredential vc.VerifiableCredential, public bool) error {
+	issuerDID, err := did.ParseDIDURL(verifiableCredential.Issuer.String())
+	if err != nil {
+		return fmt.Errorf("invalid credential issuer: %w", err)
+	}
+
+	if len(verifiableCredential.CredentialSubject) == 0 {
+		return fmt.Errorf("missing credentialSubject")
+	}
+
+	// create participants list
+	participants, err := p.generateParticipants(verifiableCredential, public)
```

Comment:
I find it odd to generate a participant list when the VC is public: because there should be no participant list (or an empty one) at all when `public == true`. So don't pass `public` to `generateParticipants`: only call it when `!public`.